### PR TITLE
[vimeo] fix album extraction (issue #15704)

### LINKED ISSUE #15704: [Vimeo] Downloading Channel/Album - Unable to extract list title
---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2018.02.26*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x ] I've **verified** and **I assure** that I'm running youtube-dl **2018.02.26**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones
- [x] Checked that provided video/audio/playlist URLs (if any) are alive and playable in a browser

### What is the purpose of your *issue*?
- [x] Bug report (encountered problems with youtube-dl)

---

Attempted to download a channel/album and recieved this error. Tried both `2018.02.22` and `2018.02.26`

*Command:*

```shell
youtube-dl "https://vimeo.com/album/4786409" -v
```

*Output:*

```shell
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['https://vimeo.com/album/4786409', '-v']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2018.02.26
[debug] Python version 3.6.3 (CPython) - Darwin-17.4.0-x86_64-i386-64bit
[debug] exe versions: ffmpeg 3.4.2, ffprobe 3.4.2
[debug] Proxy map: {}
[vimeo:album] 4786409: Downloading page 1
ERROR: Unable to extract list title; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 785, in extract_info
    ie_result = ie.extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 440, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/vimeo.py", line 855, in _real_extract
    return self._extract_videos(album_id, 'https://vimeo.com/album/%s' % album_id)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/vimeo.py", line 791, in _extract_videos
    list_title = next(title_and_entries)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/vimeo.py", line 768, in _title_and_entries
    yield self._extract_list_title(webpage)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/vimeo.py", line 730, in _extract_list_title
    return self._TITLE or self._html_search_regex(self._TITLE_RE, webpage, 'list title')
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 805, in _html_search_regex
    res = self._search_regex(pattern, string, name, default, fatal, flags, group)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 796, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract list title; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

### PR DESCRIPTION
**Pull request by kayb94**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

(Mixed, took some snippets from elsewhere in youtube-dl)

### What is the purpose of your *pull request*?
- [x] Bug fix
---

### Description of your *pull request* and other information

Fixes the vimeo album playlist extraction. I added a test (from the original bug report). So, this resolves #15704.

### CODE REVIEW

**Comment by PSlava** [comment]
This fix fails to extract password protected albums! 
For example (from test) : https://vimeo.com/album/3253534

**Comment by kayb94** [comment]
As said, that's not what I tried to achieve. Please open a seperate Issue, if you haven't done yet.

Regards

**Comment by kayb94** [comment]
@dstftw Is this ready?

Regards

File: youtube_dl/extractor/vimeo.py [hunk 1]
```diff
@@ -833,9 +834,16 @@ class VimeoAlbumIE(VimeoChannelIE):
         'url': 'https://vimeo.com/album/2632481',
         'info_dict': {
             'id': '2632481',
-            'title': 'Staff Favorites: November 2013',
+            'title': 'Vimeo / Staff Favorites: November 2013',
```
Comment (dstftw) [inline youtube_dl/extractor/vimeo.py]: The title is `Staff Favorites: November 2013`.

**Comment by dstftw** [comment]
Does not work for multi page albums.
Does not work https://vimeo.com/album/2632481/sort:plays/format:thumbnail.
Does not work https://vimeo.com/album/2632481/page:2/sort:plays/format:thumbnail.

File: youtube_dl/extractor/vimeo.py [hunk 2]
```diff
@@ -861,7 +869,34 @@ def _page_url(self, base_url, pagenum):
 
     def _real_extract(self, url):
         album_id = self._match_id(url)
-        return self._extract_videos(album_id, 'https://vimeo.com/album/%s' % album_id)
+        rss_url = url + '/rss'
+
+        doc = self._download_xml(rss_url, album_id, fatal=True)
```
Comment (dstftw) [inline youtube_dl/extractor/vimeo.py]: `True` is default.

File: youtube_dl/extractor/vimeo.py [hunk 3]
```diff
@@ -861,7 +869,34 @@ def _page_url(self, base_url, pagenum):
...
+
+        playlist_title = doc.find('./channel/title').text
```
Comment (dstftw) [inline youtube_dl/extractor/vimeo.py]: Should not be fatal.

File: youtube_dl/extractor/vimeo.py [hunk 4]
```diff
@@ -861,7 +869,34 @@ def _page_url(self, base_url, pagenum):
...
+        playlist_desc_el = doc.find('./channel/description')
+        playlist_desc = None if playlist_desc_el is None else playlist_desc_el.text
+
+        entries = []
+        for it in doc.findall('./channel/item'):
+            next_title = it.find('title').text
+            next_url = xpath_text(it, 'link', fatal=False)
```
Comment (dstftw) [inline youtube_dl/extractor/vimeo.py]: `False` is default.

**Comment by kayb94** [comment]
I resolved the things you requested, could you review again please @dstftw 

Regards

**Pull request closed without merging**